### DISCUSSION TITLE
How can I get the power to grid data as an additional entity?

### DISCUSSION THREAD

**Discussion by solesia0699** [binsentsu/home-assistant-solaredge-modbus#211] [Q&A]
Hello,I couldn't locate the following entity among the 23 displayed: a sensor that expresses negative power in watts when exporting to the grid and positive power when consuming from the grid. Is it possible to add this data? I am not a programmer and have no clue how to do this. Thanks for your help.

**Comment by mpredfearn** [comment]
The Meter AC power is the sensor you want for this. I assume that you have a modbus import / export meter for your system?

![Screenshot_2023-11-16_17-34-43](https://github.com/binsentsu/home-assistant-solaredge-modbus/assets/12936891/76988d24-8d39-4ce1-8728-1a73be62a1bf)

Note the sense is inverted versus what you said - it's positive on export and negative on import.

**Comment by solesia0699** [comment]
Hi,

I have an SE8K. Currently, I am using the SolarEdge Modbus integration, not the SolarEdge Modbus multi. I can access 23 metrics.

Here is what I have for the 'solaredge_ac_power' metric:
![image](https://github.com/binsentsu/home-assistant-solaredge-modbus/assets/145588737/59d92fce-cbb4-49df-bb36-ee1224163711)

As you can see, it does not display negative values. When I compare it with the data provided by the SolarEdge Application, this metric aligns with solar production, so it does not help.

**Reply by mpredfearn** [reply]
Yeah - that sensor is the inverter AC power. Do you have a modbus import / export meter? If not, there is no way that the inverter can read this. The modbus meter has a CT clamp around the mains feed and it can then measure the current here and from that you also get the power flow. Do you have such a meter?

**Reply by solesia0699** [reply]
Ah, okay. I don't have such a meter; I rely solely on the Modbus TCP data available on the card in the inverter. Could you please explain how the official SolarEdge app measures this information and provides a live update? My concern is that the integration with the SolarEdge API only provides data every 15 minutes, which is too lengthy

**Reply by mpredfearn** [reply]
Solaredge app measures this exactly the same way. You need the meter hardware to do this (https://www.solaredge.com/en/products/metering-and-sensors/energy-meter-with-modbus-connection) specifically you need an import / export meter at the grid connection point in order for the inverter to measure this (https://knowledge-center.solaredge.com/sites/kc/files/solaredge-meter-installation-guide.pdf) otherwise, the inverter can only provide you with the data it has itself i.e. how much power is being produced - it cannot measure how much of that power is being exported / consumed because it simply has no way to measure that.

**Reply by solesia0699** [reply]
Okay, this means I have the meter hardware then. Take a look at a picture of my configuration.
![solaredge](https://github.com/binsentsu/home-assistant-solaredge-modbus/assets/145588737/a3f100ec-f377-44e9-bc40-d88e3e309335)

and this is what I can see on my app:
![solaredge app screenshot](https://github.com/binsentsu/home-assistant-solaredge-modbus/assets/145588737/485e2a6c-268d-42c2-b8de-6c1c248e0699)

So, I have an RJ45 connection to the card but cannot retrieve the required data, as I can see in the app...

**Reply by mpredfearn** [reply]
OK, cool, well that's good - yes your meter will be connected via RS485 to that RS-485-1 port on the board.

I haven't used this integration in a while, as I migrated to https://github.com/WillCodeForCats/solaredge-modbus-multi which I find better supported, but IIRC, during configuration of the integration, there is a checkbox for "Read Meter 1". Make sure that when you configure the integration, you enable this so that the integration knows to interrogate the inverter for the meter. If the meter is found successfully, then the additional entities for it will be added. I don't think there is a way to reconfigure the integration to enable this after it is added, so you'll have to remove it and re-add it to see the option (BTW https://github.com/WillCodeForCats/solaredge-modbus-multi does allow for this reconfiguration after the fact).

**Reply by solesia0699** [reply]
So I tried the solaredge-modbus-multi (after having removed the solaredge-modbus) and this now works! Thanks a lot for the help.
![image](https://github.com/binsentsu/home-assistant-solaredge-modbus/assets/145588737/4766f0be-8289-44e1-b8fd-ee9cc872079f)

This matches what I can see on the solaredge app

**Reply by mpredfearn** [reply]
Excellent :-)
